Pegasus/sources/particleworld.cpp

```diff
@@ -1,74 +1,64 @@
 #include "Pegasus/include/particleworld.hpp"
 
-pegas::ParticleWorld::ParticleWorld(unsigned int maxContacts,
-    unsigned int iterations)
-    : mResolver(iterations)
+pegasus::ParticleWorld::ParticleWorld(
+    Particles & particles,
+    ParticleForceRegistry & forceRegistry,
+    ParticleContactGenerators & contactGenerators,
+    uint32_t maxContacts,
+    uint32_t iterations)
+    : mParticles(particles)
+    , mForceRegistry(forceRegistry)
+    , mContactGenerators(contactGenerators)
+    , mContactResolver(iterations)
+    , mCalculateIterations(false)
     , mMaxContacts(maxContacts)
 {
+    mContacts.reserve(mMaxContacts);
 }
 
-void pegas::ParticleWorld::startFrame()
+void pegasus::ParticleWorld::startFrame() const
```

Comment:
Cause you don't change a variable inside the ParticleWorld, you change the reference to the object that is outside of it.